services/edus/src/main/java/app/coronawarn/datadonation/services/edus/otp/OtpController.java

```diff
@@ -59,7 +60,25 @@ public OtpController(OtpService otpService) {
       logger.warn("OTP could not be redeemed.");
     }
 
-    return new ResponseEntity<>(new OtpRedemptionResponse(otpRedemptionRequest.getOtp(), otpState),
+    return new ResponseEntity<>(new OtpRedemptionResponse(otpRedemptionRequest.getOtp(), otpState,
+        calculateStrongClientIntegrityCheck(otp)),
         httpStatus);
   }
+
+  private boolean calculateStrongClientIntegrityCheck(OneTimePassword otp) {
+    return isOtpFromIosDevice(otp) || isOtpFromValidAndroidDevice(otp);
+  }
+
+  private boolean isOtpFromIosDevice(OneTimePassword otp) {
+    return otp.getAndroidPpacBasicIntegrity() == null
+        && otp.getAndroidPpacCtsProfileMatch() == null
+        && otp.getAndroidPpacEvaluationTypeBasic() == null
+        && otp.getAndroidPpacEvaluationTypeHardwareBacked() == null;
+  }
+
+  private boolean isOtpFromValidAndroidDevice(OneTimePassword otp) {
+    return TRUE.equals(otp.getAndroidPpacBasicIntegrity())
+        && TRUE.equals(otp.getAndroidPpacCtsProfileMatch())
+        && TRUE.equals(otp.getAndroidPpacEvaluationTypeHardwareBacked());
```

Comment:
Testing if something is true with equals? Why not:


```suggestion
    return otp.getAndroidPpacBasicIntegrity()
        && otp.getAndroidPpacCtsProfileMatch()
        && otp.getAndroidPpacEvaluationTypeHardwareBacked();
```

Reply:
It's required because the Boolean (not boolean) would be cast to boolean and throw an exception if it's null

Reply:
ah! sorry... my bad